d3-array/d3-array-tests.ts

```diff
@@ -330,12 +330,23 @@ mergedArray = d3Array.merge(testArrays); // inferred type
 mergedArray = d3Array.merge<MixedObject>(testArrays); // explicit type
 // mergedArray = d3Array.merge<MixedObject>([[10, 40, 30], [15, 30]]); // fails, type mismatch
 
+// cross() ---------------------------------------------------------------------
+
+let crossed: Array<[string, number]>;
+
+crossed = d3Array.cross<string, number>(['x', 'y'], [1, 2]);
```

Comment:
I have added/modified a couple of tests.

At this point, I am only focusing on the tests directly related to the changed signatures. I.e. for this PR, to be timely, I do not have the cycles to expand the testing of d3-array beyond what is currently in place.
Thx. T.